torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -140,6 +140,8 @@ class TORCH_API ProcessGroupNCCL : public ProcessGroup {
     // Helper function that returns True if the WorkNCCL object has timed out
     // and False otherwise.
     bool timedOut();
+    // Overloaded version of the timed out function.
+    bool timedOut(std::chrono::milliseconds timeout);
```

Comment:
could we just use `timedOut` and allow an optional argument for timeout?

Reply:
We could use `c10::optional`.